File: code_challenge_solution/javascript/38 Counts occurrences of characters in a Given String.js
let str = "Hello World!";
let charFrequency = {};

for (let i = 0; i < str.length; i++) {
  let char = str.charAt(i);
  if (charFrequency[char]) {
    charFrequency[char]++;
  } else {
    charFrequency[char] = 1;
  }
}

console.log(charFrequency);

File: code_challenge_solution/java/38 Counts occurrences of characters in a Given String.java
import java.util.HashMap;
import java.util.Map;

public class CharacterFrequency {
    public static void main(String[] args) {
        String str = "Hello world!";
        countCharacterFrequency(str);
    }

    public static void countCharacterFrequency(String str) {
        Map<Character, Integer> frequencyMap = new HashMap<>();

        for (int i = 0; i < str.length(); i++) {
            char ch = str.charAt(i);

            if (frequencyMap.containsKey(ch)) {
                frequencyMap.put(ch, frequencyMap.get(ch) + 1);
            } else {
                frequencyMap.put(ch, 1);
            }
        }

        for (Map.Entry<Character, Integer> entry : frequencyMap.entrySet()) {
            System.out.println(entry.getKey() + " : " + entry.getValue());
        }
    }
}